clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/common/AzureUtilities.groovy

```diff
@@ -112,6 +113,30 @@ class AzureUtilities {
     resourceId.split(PATH_SEPARATOR).last()
   }
 
+  // id = "/subscriptions/***-***-***/resourceGroups/***/providers/Microsoft.Network/networkInterfaces/nic1/ipConfigurations/ipconfig1"
+  static String getResourceNameFromId(String id) {
```

Comment:
This method needs to be generic and not hard-coded to just a single provider. What if we need to get the name of a resource that's from a different provider?